experimental/http_logger/http_logger.go

```diff
@@ -0,0 +1,100 @@
+package http_logger
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/experimental/e2e/fixture"
+	"github.com/grafana/grafana-plugin-sdk-go/experimental/e2e/storage"
+	"github.com/grafana/grafana-plugin-sdk-go/experimental/e2e/utils"
+)
+
+const (
+	// PluginHARLogEnabledEnv is a constant for the GF_PLUGIN_HAR_LOG_ENABLED environment variable used to enable HTTP request and responses in HAR format for debugging purposes.
+	PluginHARLogEnabledEnv = "GF_PLUGIN_HAR_LOG_ENABLED"
+	// PluginHARLogPathEnv is a constant for the GF_PLUGIN_HAR_LOG_PATH environment variable used to specify a path to store HTTP request and responses in HAR format for debugging purposes.
+	PluginHARLogPathEnv = "GF_PLUGIN_HAR_LOG_PATH"
+)
+
+type HttpLogger struct {
+	enabled bool
+	proxied http.RoundTripper
+	fixture *fixture.Fixture
+}
+
+func NewHTTPLogger(pluginID string, proxied http.RoundTripper) *HttpLogger {
+	if pluginID == "" {
+		panic("pluginID cannot be empty")
+	}
+
+	path, ok := os.LookupEnv(PluginHARLogPathEnv)
+	if !ok {
+		path = createTemp(pluginID)
+	}
+
+	enabled := false
+	if v, ok := os.LookupEnv(PluginHARLogEnabledEnv); ok && v == "true" {
+		backend.Logger.Info("HTTP HAR Logging enabled", "pluginID", pluginID, "path", path)
+		enabled = true
+	}
+
+	s := storage.NewHARStorage(path)
+	if err := s.Load(); err != nil {
```

Comment:
Am I reading this right? If there is an error, then run `s.Init()`?

Reply:
@srclosson apparently i forgot to comment here, but i attempted to simplify this in [e677872](https://github.com/grafana/grafana-plugin-sdk-go/pull/471/commits/e67787269741c14de0b642fb56aef24eb3a6e36e)